compiler/resolution/initializerRules.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2004-2016 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "initializerRules.h"
+
+#include "astutil.h"
+#include "expr.h"
+#include "stmt.h"
+#include "stringutil.h"
+#include "symbol.h"
+#include "type.h"
+
+// Helper file for verifying the rules placed on initializers, and providing
+// the extra functionality associated with them.
+
+void temporaryInitializerFixup(CallExpr* call) {
```

Comment:
I don't really understand the business logic regarding the checks for "init" and "meme" in this function.
It seems to me that the caller has constrained the "syntax" of the call.  It seems like an internal error if
the call is not "init" and it seems like "meme" is expected to be in a particular place.  Perhaps either a)
assume that b) assume/verify.  Quietly skipping things seems brittle.